src/Integrations/Integrations/TYPO3/TYPO3Integration.php

```diff
@@ -0,0 +1,254 @@
+<?php
+
+namespace DDTrace\Integrations\TYPO3;
+
+use DDTrace\SpanData;
+use DDTrace\Integrations\Integration;
+use DDTrace\Tag;
+use DDTrace\Type;
+
+class TYPO3Integration extends Integration
+{
+    const NAME = 'typo3';
+
+    /**
+     * @var string
+     */
+    private $serviceName = '';
+
+    /**
+     * @return string The integration name.
+     */
+    public function getName()
+    {
+        return self::NAME;
+    }
+
+    /**
+     * @return string
+     */
+    public function getServiceName()
+    {
+        if (empty($this->serviceName)) {
+            $this->serviceName = \ddtrace_config_app_name('typo3');
+        }
+
+        return $this->serviceName;
+    }
+
+    /**
+     * @return int
+     */
+    public function init()
+    {
+        if (!self::shouldLoad(self::NAME)) {
+            return Integration::NOT_LOADED;
+        }
+
+        $rootSpan = \DDTrace\root_span();
+
+        if (null === $rootSpan) {
+            return Integration::NOT_LOADED;
+        }
+
+        $integration = $this;
+
+        $setCommonValues = function (SpanData $span) {
+            $this->setCommonValues($span);
+        };
+
+        \DDTrace\trace_method(
+            \TYPO3\CMS\Frontend\Http\Application::class,
```

Comment:
`*::class` is not compatible with PHP 5.4. So you should either load this extension only on 5.5+ (in `IntegrationsLoader` - also only 7.x+ is perfectly acceptable if not preferable for us) or replace all usage of `::class`.